### DISCUSSION TITLE
CONSULT-FIND doesn't show previews when called non-interactively or with optional arguments

### DISCUSSION THREAD

**Discussion by hapst3r** [minad/consult#1143] [Q&A] [answered]
Hej minad,

out of the box, CONSULT-FIND does not show a preview, but it is possible to make it do so using CONSULT-CUSTOMIZE with the following invocation: `(consult-customize consult-find :state (consult--file-preview))`. 

I love this functionality, and it works just as I'd expect it to when calling CONSULT-FIND interactively via shortcut or EXECUTE-EXTENDED-COMMAND. However, it does not work when CONSULT-FIND is called non-interactively or with optional arguments, which I'd love to be able to in the following two use cases:

1. Prompt for directory by default by means of a (probably very badly programmed) function which passes the prefix argument required to do so (to mimic `C-u M-x consult-find`.

```
(defun or/consult-find-with-dir-selection ()
    "Call CONSULT-FIND while prompting for searched directory/directories."
    (interactive)
    (let ((current-prefix-arg 4))
      (call-interactively #'consult-find)))
```

2. I built a transient that calls CONSULT-FIND with both DIR and INITIAL arguments to query my denote notes. I could use CONSULT-DENOTE-FIND for this, but I still like the idea to have INITIAL narrowing on a specific DIR (less typing in the end of the day). The following function does not yield previews even though it should in my understanding (both with config and in `emacs -Q`):

`(consult-find "$DIR" "$INITIAL)`

I tried reading the code that implements CONSULT-CUSTOMIZE and experimented with calling it before programmatically calling CONSULT-FIND with optional arguments, but I struggle understanding the code and the experiments were not successful. Code below:

```
(defun or/consult-query-notes-emacs ()
  (interactive)
  (consult-customize consult-find :state (consult--file-preview))
  (consult-find (concat or/denote-base-directory "computer") "==emacs "))
```

Thanks for your awesome contribution to the emacs community :)

**Comment by minad** [comment, marked as answer]
`consult-customize` relies on `this-command`. This means let-binding `this-command` should work.

**Reply by hapst3r** [reply]
Thank you so much your input. It took me some time to understand how `this-command` works in general and in the context of `consult-customize`. You were correct, let-binding `this-command` is the solution.

# this works
```lisp
(defun or/consult-query-notes-emacs ()
  (interactive)
  (let ((this-command 'consult-find))
    (consult-find (concat or/denote-base-directory "computer") "==emacs ")))
```

# this doesn't work
```lisp
(defun or/consult-query-notes-emacs ()
  (interactive)
    (consult-find (concat or/denote-base-directory "computer") "==emacs "))
```
